### PR TITLE
Created signup page

### PR DESCRIPTION
**Pull request by atmiguel**

### CODE REVIEW

File: templates/login_or_signup.tmpl [hunk 1]
```diff
@@ -5,17 +5,105 @@
...
+                                />
+                                <label>Name</label>
+                                <span class="validation-message mui--text-accent-secondary"></span>
+                            </div>
+
+                            <div class="mui-textfield mui-textfield--float-label">
+                                <input
+                                    type="email"
+                                    name="emailAddress"
+                                    maxlength="128"
```
Comment (zingales) [inline templates/login_or_signup.tmpl:64]: why add max? is there some restriction based on some tech we use?

Comment (atmiguel) [reply templates/login_or_signup.tmpl:64]: yeah i don't have a strong reason. i could be convinced to remove. my only thought was that it seems unlikely that anything over this length would be a real user

File: static/js/login_or_signup.js [hunk 2]
```diff
@@ -1 +1,107 @@
-alert('loaded js');
+var getInputField = function($form, field) {
+    return $form.find('[name="' + field + '"]');
+};
+
+var getInputFields = function($form, fields) {
+    return fields.map(
+        field => getInputField($form, field));
```
Comment (atmiguel) [inline static/js/login_or_signup.js]: add parens

File: static/js/login_or_signup.js [hunk 3]
```diff
@@ -1 +1,107 @@
...
+    $validationMessage.text(validationMessage);
+};
+
+var populateValidationMessages = function($form, fields) {
+    getInputFields($form, fields).forEach(
+        $field => populateValidationMessage($field));
+};
+
+var touchAllFields = function($form, fields) {
+    getInputFields($form, fields).forEach($field => {
```
Comment (atmiguel) [inline static/js/login_or_signup.js]: newline

File: static/js/login_or_signup.js [hunk 4]
```diff
@@ -1 +1,107 @@
...
+    var emailAddressField = 'emailAddress';
+    var passwordField = 'password';
+
+    var fields = [
+        displayNameField,
+        emailAddressField,
+        passwordField,
+    ];
+
+    getInputFields($signupForm, fields).forEach($field => {
```
Comment (atmiguel) [inline static/js/login_or_signup.js]: newline

File: static/js/login_or_signup.js [hunk 5]
```diff
@@ -1 +1,107 @@
...
+
+        // continuously update validation message after failed submission
+        $field.on('input', event => {
+            if (submitHasBeenClicked) {
+                populateValidationMessage($field);
+            }
+        });
+
+        if (field !== passwordField) {
+            // restrict inital space character
```
Comment (atmiguel) [inline static/js/login_or_signup.js]: typo

File: static/js/login_or_signup.js [hunk 6]
```diff
@@ -1 +1,107 @@
...
+            $field.keypress((event) => {
+                var value = $field.val();
+                var trimmedValue = $.trim(value);
+
+                if (checkKeypressIsSpace(event) && trimmedValue.length === 0) {
+                    return false;
```
Comment (atmiguel) [inline static/js/login_or_signup.js]: leave comment explaining

File: static/js/login_or_signup.js [hunk 7]
```diff
@@ -1 +1,107 @@
...
+            $field.blur(() => {
+                var value = $field.val();
+                var trimmedValue = $.trim(value);
+
+                $field.val(trimmedValue);
+            });
+        }
+    });
+
+    $signupForm.find('button').click(function() {
```
Comment (atmiguel) [inline static/js/login_or_signup.js]: arrow function

**Review by zingales — DISMISSED** [review]

**Pull request merged**